### DISCUSSION TITLE
Path prefix when behind a proxy

### DISCUSSION THREAD

**Discussion by jgomezb** [apache/superset#14440] [Ideas]
### Context
Currently, in the company I work for, we are evaluating the use of Apache Superset for visualizations on top of the current Druid infrastructure we have. This would be a service integrated in the company infrastructure, which requires to go through some gateway that routes urls with a prefix path /yourservice/ to the specific service. After trying every single possibility and solution proposed out there (like blueprints or APPLICATION_ROOT/SCRIPT_NAME middleware), I found out this is difficult because Flask on which Apache Superset is based only controls the urls in templates, but then:
- Embedded static urls in templates (/static/assets/).
- FAVICONS configuration parameter is hardcoded to /static/assets/images/favicon.png.
- js_manifest and css_manifest do not allow a prefix.
- Api requests (/api/v1/dataset) in react through SupersetClient do not allow a prefix.
- Some urls in dependencies like /superset/csrf_token in SupersetClientClass.getCSRFToken point directly to the Flask route.

All of this means there is no easy way to use Superset behind a proxy, or at least I did not find it possible (I may be missing something here).

### Questions

My question is: would this be a feature that could be added to the project? If so, I could try to send a PR adding a SUPERSET_WEBSERVER_PREFIX (or something) to:
+ Adjust static urls in templates from context passed from the view that allows the prefix.
+ Adjust FAVICONS after importing the superset_config.py file to account for the prefix.
+ js_manifest and css_manifest would use the app.config['SUPERSET_WEBSERVER_PREFIX'] to provide asset urls.
+ SupersetClient receives bootstrap_data with base prefix, or another option is to expose in templates a url_for (similar to Flask's one) that maps the routes from bootstrap_data to be used by api calls to get the final url.
+ Use the baseUrl instead of just host and protocol in SupersetClientClass for getCSRFToken.

The last two probably would require a PR to superset-ui/core.

Another possibility is to use window.location.href as base for API requests, or any other request coming from the frontend (other projects in the company follow that approach successfully).

Am I missing any url in the list? Would a change like this be accepted as a contribution?

**Comment by jgomezb** [comment]
This is not getting a lot of attention, does that mean it is not a useful feature that would fit in the roadmap?

**Reply by nytai** [reply]
we generally recommend using a subdomain for superset because of this issue. You're welcome to give this a try, the static assets should already have support for a host/prefix to enable hosting assets on a CDN. There's just a lot to cover and then ideally there would be tests so no one accidentally breaks this when adding a new route/request.
